src/renderer.zig

```diff
@@ -606,6 +610,312 @@ pub const MenuBackGround = struct {
+		.{0.000029353260, 0.000010600000, 0.000000000000},
+		.{0.000020673830, 0.000007465700, 0.000000000000},
+		.{0.000014559770, 0.000005257800, 0.000000000000},
+		.{0.000010253980, 0.000003702900, 0.000000000000},
+		.{0.000007221456, 0.000002607800, 0.000000000000},
+		.{0.000005085868, 0.000001836600, 0.000000000000},
+		.{0.000003581652, 0.000001293400, 0.000000000000},
+		.{0.000002522525, 0.000000910930, 0.000000000000},
+		.{0.000001776509, 0.000000641530, 0.000000000000},
+		.{0.000001251141, 0.000000451810, 0.000000000000}
+	};
+
+	fn plankRadiance(temperature: f64, wavelength: f64) f64 {
+		const meters = wavelength * 1e-9;
+		const c2 = speedOfLight * speedOfLight;
+		const meter5 = meters * meters * meters * meters * meters;
+		return (2 * planck * c2) / (meter5 * (@exp(planck * speedOfLight / (meters * boltzmann * temperature)) - 1));
+	}
+
+	fn getStarColor(temperature: f64, magnitude: f64) Vec3d {
+		var total: Vec3d = @splat(0);
+		for (0..wavelengths.len) |i| {
+			const wavelength: f64 = @floatFromInt(i * 5 + 360);
+			const spectrum = plankRadiance(temperature, wavelength);
+			total += @as(Vec3d, @splat(spectrum * 5)) * wavelengths[i];
+		}
+
+		total /= @splat(@reduce(.Max, total));
+
+		const conv = [_]Vec3d{
+			.{ 3.2406, -0.9689, 0.0557 },
+			.{ -1.5372, 1.8758, -0.2040 },
+			.{ -0.4986, 0.0415, 1.0570 },
+		};
+
+		var rgb = conv[0] * @as(Vec3d, @splat(total[0])) + conv[1] * @as(Vec3d, @splat(total[1])) + conv[2] * @as(Vec3d, @splat(total[2]));
+
+		const threshold = 0.0031308;
+		const a = @as(Vec3d, @splat(1.055));
+		const b = @as(Vec3d, @splat(12.92));
+		const gamma = 1.0 / 2.4;
+
+		const mask = rgb > @as(Vec3d, @splat(threshold));
+		const corrected = a * Vec3d {
+			std.math.pow(f64, rgb[0], gamma),
+			std.math.pow(f64, rgb[1], gamma),
+			std.math.pow(f64, rgb[2], gamma),
+		} - @as(Vec3d, @splat(0.055));
+		rgb = @select(f64, mask, corrected, rgb * b);
+
+		const light = std.math.pow(f64, 10.0, -0.4 * magnitude);
+
+		rgb *= @as(Vec3d, @splat(light));
+
+		const m = @reduce(.Max, rgb);
+		if (m > 1.0) {
+			rgb /= @as(Vec3d, @splat(m));
+		}
+
+		return rgb;
+	}
+
+	fn init() void {
+		starShader = Shader.initAndGetUniforms("assets/cubyz/shaders/skybox/star.vs", "assets/cubyz/shaders/skybox/star.fs", "", &starUniforms);
+		starShader.bind();
+
+		var starData: [NUM_STARS * 6]f32 = undefined;
+
+		var starRandom = std.Random.DefaultPrng.init(0);
+
+		for (0..NUM_STARS) |i| {
+			const pos = getStarPos(starRandom.random());
+
+			const radius: f64 = starRandom.random().floatExp(f64) * 4 + 0.2;
+
+			const temperature: f64 = (@abs(starRandom.random().floatNorm(f64) * 3000.0 + 5000.0) + 1000.0) / 5772.0;
+
+			const luminosity = 4.0 * std.math.pi * radius * radius * temperature * temperature * temperature * temperature;
+
+			const flux = luminosity / (4.0 * std.math.pi * 1.36e+7 * vec.length(pos) * vec.length(pos));
+
+			const magnitude = -2.5 * @log10(flux) - 26.83;
```

Comment:
Do we really need to drag around all this physics, if we could just choose a random brightness to begin with?